### ISSUE TITLE
Why Firebase Messaging not initializing?

### ISSUE THREAD

**Issue by XLIVE99** [firebase/firebase-unity-sdk#882]
### [REQUIRED] Please fill in the following fields:

  * Unity editor version: 2021.3.18f1
  * Firebase Unity SDK version: 11.4.0 and 11.5.0
  * Source you installed the SDK: https://firebase.google.com/docs/cloud-messaging/unity/client?hl=tr#before_you_begin
  * Problematic Firebase Component: Messaging
  * Other Firebase Components in use: Analytics, Crashlytics, Remote Config
  * Additional SDKs you are using: Facebook, AdMob
  * Platform you are using the Unity editor on: Mac
  * Platform you are targeting: iOS
  * Scripting Runtime: IL2CPP
  * Pre-built SDK from the [website](https://firebase.google.com/download/unity)

### [REQUIRED] Please describe the question here:

Push notification is not working on iOS in this project. I also created an empty project and push notification was working fine there. So I looked for differences and found this:

### **Not working log file different section:**

Application Version : 1.1.6
FirebaseSettings:VersionControlRemoteConfig()

2023-10-12 17:17:34.740404+0300 PaintByNumberMaster[5217:3476970] 10.13.0 - [FirebaseCore][I-COR000005] No app has been configured yet.
2023-10-12 17:17:34.740564+0300 PaintByNumberMaster[5217:3476970] 10.13.0 - [FirebaseCore][I-COR000003] The default Firebase app has not yet been configured. Add \`FirebaseApp.configure()\` to your application initialization. This can be done in in the App Delegate's application(_:didFinishLaunchingWithOptions:)\` (or the \`@main\` struct's initializer in SwiftUI). Read more: https://goo.gl/ctyzm8.
2023-10-12 17:17:34.740723+0300 PaintByNumberMaster[5217:3476970] 10.13.0 - [FirebaseCore][I-COR000005] No app has been configured yet.
2023-10-12 17:17:34.747348+0300 PaintByNumberMaster[5217:3476672] [Firebase/Crashlytics] Version 10.13.0
2023-10-12 17:17:34.754744+0300 PaintByNumberMaster[5217:3476968] 10.13.0 - [FirebaseAnalytics][I-ACS023007] Analytics v.10.13.0 started
2023-10-12 17:17:34.755104+0300 PaintByNumberMaster[5217:3476968] 10.13.0 - [FirebaseAnalytics][I-ACS023008] To enable debug logging set the following application argument: -FIRAnalyticsDebugEnabled (see http://goo.gl/RfcP7r)
2023-10-12 17:17:34.761018+0300 PaintByNumberMaster[5217:3477011] 10.13.0 - [FirebaseMessaging][I-FCM001000] FIRMessaging Remote Notifications proxy enabled, will swizzle remote notification receiver handlers. If you'd prefer to manually integrate Firebase Messaging, add "FirebaseAppDelegateProxyEnabled" to your Info.plist, and set it to NO. Follow the instructions at:
https://firebase.google.com/docs/cloud-messaging/ios/client#method_swizzling_in_firebase_messaging
to ensure proper integration.
2023-10-12 17:17:34.767194+0300 PaintByNumberMaster[5217:3477011] 10.13.0 - [FirebaseRemoteConfig][I-RCN000072] New config database created. Resetting user defaults.
Firebase init
FirebaseSettings:<FirebaseInitialize>b__12_0(Task\`1)
Firebase.Extensions.<>c__DisplayClass4_1\`1:<ContinueWithOnMainThread>b__1()
Firebase.<>c__DisplayClass5_0\`1:<RunAsync>b__0()
Firebase.ExceptionAggregator:Wrap(Action)
Firebase.Dispatcher:PollJobs()
Firebase.Platform.FirebaseHandler:Update()

Called task from queue
FirebaseSettings:<FirebaseInitialize>b__12_0(Task\`1)
Firebase.Extensions.<>c__DisplayClass4_1\`1:<ContinueWithOnMainThread>b__1()
Firebase.<>c__DisplayClass5_0\`1:<RunAsync>b__0()
Firebase.ExceptionAggregator:Wrap(Action)
Firebase.Dispatcher:PollJobs()
Firebase.Platform.FirebaseHandler:Update()

Fetching data...
FirebaseSettings:FetchRemoteConfig(Action)
Firebase.Extensions.<>c__DisplayClass0_1:<ContinueWithOnMainThread>b__1()
Firebase.<>c__DisplayClass5_0\`1:<RunAsync>b__0()
Firebase.ExceptionAggregator:Wrap(Action)
Firebase.Dispatcher:PollJobs()
Firebase.Platform.FirebaseHandler:Update()

------TOKEN-------: daHLvb44U0XWvuuQQA5F8V:APA91bHf5vrp1_8SeWH1v-rReh1qoWZ8_QADIy5yK_UFiCKaKkhE8z2Pf2P8YqAC1ImbkOXsQSoOpLQkh4zYqUDwZNp-sMCKHldj4YqoqtmONzd0A1Wg7lVbz99cqFdSQu1TZmKczq9Y
Firebase.ExceptionAggregator:Wrap(Action)
Firebase.AppUtil:PollCallbacks()
Firebase.Platform.FirebaseHandler:Update()

Thread Performance Checker: Thread running at QOS_CLASS_USER_INTERACTIVE waiting on a thread without a QoS class specified. Investigate ways to avoid priority inversions
PID: 5217, TID: 3476672

### **Working log file different section:**

2023-10-12 17:37:56.048022+0300 test[5267:3488152] 10.15.0 - [FirebaseCore][I-COR000005] No app has been configured yet.
2023-10-12 17:37:56.048099+0300 test[5267:3488152] 10.15.0 - [FirebaseCore][I-COR000003] The default Firebase app has not yet been configured. Add \`FirebaseApp.configure()\` to your application initialization. This can be done in in the App Delegate's application(_:didFinishLaunchingWithOptions:)\` (or the \`@main\` struct's initializer in SwiftUI). Read more: https://goo.gl/ctyzm8.
2023-10-12 17:37:56.048136+0300 test[5267:3488152] 10.15.0 - [FirebaseCore][I-COR000005] No app has been configured yet.
2023-10-12 17:37:56.057179+0300 test[5267:3487918] <SKPaymentQueue: 0x283a50110>: No observers found that respond to "paymentQueue:shouldAddStorePayment:forProduct:", will not check for purchase intents
2023-10-12 17:37:56.111067+0300 test[5267:3488151] 10.15.0 - [FirebaseAnalytics][I-ACS023007] Analytics v.10.15.0 started
2023-10-12 17:37:56.111124+0300 test[5267:3488151] 10.15.0 - [FirebaseAnalytics][I-ACS023008] To enable debug logging set the following application argument: -FIRAnalyticsDebugEnabled (see http://goo.gl/RfcP7r)
2023-10-12 17:37:56.118206+0300 test[5267:3488249] 10.15.0 - [FirebaseMessaging][I-FCM001000] FIRMessaging Remote Notifications proxy enabled, will swizzle remote notification receiver handlers. If you'd prefer to manually integrate Firebase Messaging, add "FirebaseAppDelegateProxyEnabled" to your Info.plist, and set it to NO. Follow the instructions at:
https://firebase.google.com/docs/cloud-messaging/ios/client#method_swizzling_in_firebase_messaging
to ensure proper integration.
2023-10-12 17:37:56.119735+0300 test[5267:3487918] FCM: Initialize Firebase Messaging
FCM: Initialize Firebase Messaging
2023-10-12 17:37:56.120537+0300 test[5267:3487918] FCM: Using FCM senderID 84848263959
FCM: Using FCM senderID 84848263959
2023-10-12 17:37:56.120920+0300 test[5267:3487918] FCM: Retrieve registration token
FCM: Retrieve registration token
Received Registration Token: c6uRLuE4ikCHv5IUgrr2u4:APA91bHEoZoe42rwtyzsa_sDLgv52B5uhVa5nwGsMbPtGdO6T3g5nGAHiz6G8fnGX91qb27nsERjvG2RRykJx2c2CDfB7ISKY8L7qP-F7z1gPWcC4W7801FYDaBxO_8ZLLpeu4MDnQzg
FirebaseMessageListener:OnTokenReceived(Object, TokenReceivedEventArgs)
Firebase.ExceptionAggregator:Wrap(Action)
Firebase.AppUtil:PollCallbacks()
Firebase.Platform.FirebaseHandler:Update()

2023-10-12 17:37:56.124789+0300 test[5267:3487918] FCM: new registration token received.
FCM: new registration token received.
2023-10-12 17:37:56.124881+0300 test[5267:3487918] FCM: Retrieve registration token
FCM: Retrieve registration token
-> applicationWillResignActive()
2023-10-12 17:37:56.453199+0300 test[5267:3488152] 10.15.0 - [FirebaseAnalytics][I-ACS800023] No pending snapshot to activate. SDK name: app_measurement
2023-10-12 17:37:56.585179+0300 test[5267:3488249] 10.15.0 - [FirebaseAnalytics][I-ACS023012] Analytics collection enabled
2023-10-12 17:37:56.585270+0300 test[5267:3488249] 10.15.0 - [FirebaseAnalytics][I-ACS023220] Analytics screen reporting is enabled. Call Analytics.logEvent(AnalyticsEventScreenView, parameters: [...]) to log a screen view event. To disable automatic screen reporting, set the flag FirebaseAutomaticScreenReportingEnabled to NO (boolean) in the Info.plist
2023-10-12 17:37:58.618551+0300 test[5267:3487918] FCM: new registration token received.
FCM: new registration token received.
2023-10-12 17:37:58.618854+0300 test[5267:3487918] FCM: Retrieve registration token
FCM: Retrieve registration token
-> applicationDidBecomeActive()
Received Registration Token: dPKFWa82TElkvW-Yi8cWT7:APA91bGeZrAVHNHki3VMAQMprBwBEW9N3so78Hze-o12_NQe-Hjc5PInL47Vbs4wVnMZSKBQ65R6kRDdOVs-vLG7HRSisqD86P0KbcpqyoyDfC2WFPaFj_A8CidYkZsNUQRYfPkgj_4C
FirebaseMessageListener:OnTokenReceived(Object, TokenReceivedEventArgs)
Firebase.ExceptionAggregator:Wrap(Action)
Firebase.AppUtil:PollCallbacks()
Firebase.Platform.FirebaseHandler:Update()

Thread Performance Checker: Thread running at QOS_CLASS_USER_INTERACTIVE waiting on a thread without a QoS class specified. Investigate ways to avoid priority inversions
PID: 5267, TID: 3487918

As you can see, working one has "FCM: Initialize Firebase Messaging" this log while not working one doesn't have this log. I think this shows there is problem but I couldn't find what might cause this. I tried both 11.4.0 and 11.5.0 firebase versions, resulted in same.

### Full log files:

[NotWork.txt](https://github.com/firebase/firebase-unity-sdk/files/12885951/NotWork.txt)

[Works.txt](https://github.com/firebase/firebase-unity-sdk/files/12885955/Works.txt)

Edit: After replacing main project firebase and .plist with empty project firebase and .plist file. Still no initialize log but everything worked but with empty project's bundle id. We will check on firebase console side for any errors.

**Comment by paulinon** [comment]
Hi @XLIVE99,

In order to identify what's causing this to happen in your project, could you provide a [minimal, reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of your implementation? Could you also share the latest logs after adding the following line in your `Start()` function:

`Firebase.FirebaseApp.LogLevel = Firebase.LogLevel.Debug;`

**Comment by XLIVE99** [comment]
Hi,

I managed it to work but I don't know why it worked, me and my colleagues are testing it a bit more to be sure we solved this issue. We will test it more on monday. Here is the steps I have done so far if anyone needs helping:

1. Recreate apple key and update Firebase APNs Authentication Key with the created key.
2. Add new ios app to the same Firebase project with slightly different bundle id.
3. Test the same project with the newly created app's .plist file, don't forget to switch bundle id.
4. If it works, switch bundle id to the main app's bundle id (and don't forget to change .plist file).
5. Check if the main app started working. If it works you are free to delete the newly created ios app.

I thought just doing the step 1 will solve the issue but it didn't. I am suspecting there might be problem with Firebase server side, maybe there were errors while updating the app and didn't tell the client. But these are my guesses. We will inform further if the problem occurs on my colleagues.
